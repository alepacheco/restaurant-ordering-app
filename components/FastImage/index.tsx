import React, { useState, useEffect } from 'react';
import * as FileSystem from 'expo-file-system';
import { Image } from 'react-native';

const cyrb53 = function(str, seed = 0) {
  let h1 = 0xdeadbeef ^ seed,
    h2 = 0x41c6ce57 ^ seed;
  for (let i = 0, ch; i < str.length; i++) {
    ch = str.charCodeAt(i);
    h1 = Math.imul(h1 ^ ch, 2654435761);
    h2 = Math.imul(h2 ^ ch, 1597334677);
  }
  h1 =
    Math.imul(h1 ^ (h1 >>> 16), 2246822507) ^
    Math.imul(h2 ^ (h2 >>> 13), 3266489909);
  h2 =
    Math.imul(h2 ^ (h2 >>> 16), 2246822507) ^
    Math.imul(h1 ^ (h1 >>> 13), 3266489909);
  return 4294967296 * (2097151 & h2) + (h1 >>> 0);
};

const defaultUrl =
  'https://storage.googleapis.com/barapp-data-images/default-dish.jpg';
export const FastImage: React.FC<{ url: string }> = ({ url, ...options }) => {
  const [uri, setUri] = useState(url || defaultUrl);

  useEffect(() => {
    (async () => {
      const { exists } = await FileSystem.getInfoAsync(
        FileSystem.cacheDirectory + cyrb53(url)
      );
      if (exists) {
        setUri(FileSystem.cacheDirectory + cyrb53(url));
      }
    })();
  }, [url]);

  useEffect(() => {
    (async () => {
      const downloadResumable = FileSystem.createDownloadResumable(
        url,
        FileSystem.cacheDirectory + cyrb53(url)
      );
      try {
        const { uri } = await downloadResumable.downloadAsync();

        setUri(uri);
      } catch (e) {
        setUri(defaultUrl);
      }
    })();
  }, [url]);

  return <Image resizeMode="cover" source={{ uri }} {...options} />;
};
