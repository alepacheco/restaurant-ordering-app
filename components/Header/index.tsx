import React from 'react';
import styled from 'styled-components/native';

const Wrapper = styled.SafeAreaView``;

const Title = styled.Text`
  color: ${props => props.theme.textColor};
  font-weight: bold;
  margin: 18px 0 12px 12px;
  font-size: 24px;
`;

export const Header: React.FC<{ title: string }> = ({ title }) => {
  return (
    <Wrapper>
      <Title>{title}</Title>
    </Wrapper>
  );
};
