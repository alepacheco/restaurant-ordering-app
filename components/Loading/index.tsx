import React, { useState, useEffect } from 'react';
import styled from 'styled-components/native';
import LottieView from 'lottie-react-native';

const Wrapper = styled.SafeAreaView`
  display: flex;
  background-color: ${props => props.theme.contrast1};
  height: 100%;
  width: 100%;
  align-items: center;
  margin: auto;
`;

const CenterMiddle = styled.View`
  align-items: center;
  margin: auto;
`;

const LoadingText = styled.Text`
  ${props => `color: ${props.theme.textColor};`}
  text-align: center;
  font-size: 24px;
`;
const StyledLottieView = styled(LottieView)`
  height: 160px;
  width: 160px;
`;

export const Loading: React.FC<{}> = () => {
  const types = {
    default: require('../../assets/animations/loading.json') as string,
  };

  const selectedType = types.default;

  return (
    <Wrapper>
      <CenterMiddle>
        <LoadingText>Loading</LoadingText>
        <StyledLottieView source={selectedType} speed={1} loop autoPlay />
      </CenterMiddle>
    </Wrapper>
  );
};
