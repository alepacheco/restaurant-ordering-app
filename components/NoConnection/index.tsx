import React, { useState, useEffect } from 'react';
import styled from 'styled-components/native';
import LottieView from 'lottie-react-native';

const Wrapper = styled.SafeAreaView`
  display: flex;
  ${props => `background-color: ${props.theme.color};`}
  height: 100%;
  width: 100%;
  align-items: center;
  margin: auto;
`;

const CenterMiddle = styled.View`
  align-items: center;
  margin: auto;
`;

const LoadingText = styled.Text`
  ${props => `color: ${props.theme.textColor};`}
  text-align: center;
  font-size: 24px;
`;
const StyledLottieView = styled(LottieView)`
  height: 160px;
  width: 160px;
`;

export const NoConnection: React.FC<{ type?: 'drinks' }> = ({ type }) => {
  return (
    <Wrapper>
      <CenterMiddle>
        <StyledLottieView
          source={require('../../assets/animations/no-connection.json')}
          speed={1}
          loop={true}
          autoPlay
        />
        <LoadingText>You are not connected to the internet!</LoadingText>
        <LoadingText>Retry</LoadingText>
      </CenterMiddle>
    </Wrapper>
  );
};
