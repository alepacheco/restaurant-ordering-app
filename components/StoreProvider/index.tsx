import React from 'react';
import {
  StoreProvider as EasyPeasyStoreProvider,
  createStore,
} from 'easy-peasy';
import store from './store';

const storeInstance = createStore(store);

export const StoreProvider: React.FC<{}> = ({ children }) => (
  <EasyPeasyStoreProvider store={storeInstance}>
    {children}
  </EasyPeasyStoreProvider>
);
