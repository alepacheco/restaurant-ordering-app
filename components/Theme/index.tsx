import React from 'react';
import { AppearanceProvider, useColorScheme } from 'react-native-appearance';
import { ThemeProvider } from 'styled-components/native';
import { theme } from './theme';

declare module 'styled-components' {
  export interface DefaultTheme {
    colorScheme: 'dark' | 'light' | 'no-preference';
    color: string;
    textColor: string;

    contrast0_5: string;
    contrast0: string;
    contrast1: string;
    contrast2: string;
    contrast3: string;
    contrast4: string;
    contrast5: string;
    contrast6: string;
    contrast7: string;
    contrast8: string;
    contrast9: string;
    contrast10: string;
  }
}

export const Theme = ({ children }: { children: any }) => {
  return (
    <AppearanceProvider>
      <ThemeProvider
        theme={{
          colorScheme: useColorScheme(),
          ...theme({ colorScheme: useColorScheme() }),
        }}>
        {children}
      </ThemeProvider>
    </AppearanceProvider>
  );
};
