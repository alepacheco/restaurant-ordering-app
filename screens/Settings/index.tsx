import React from 'react';
import { Text } from 'react-native';
import { Header } from 'components/Header';
import styled from 'styled-components/native';

const Wrapper = styled.View`
  background-color: ${props => props.theme.contrast1};
  display: flex;
`;

export const Settings: React.FC<{}> = ({}) => {
  return (
    <Wrapper>
      <Header title="Need help?" />
    </Wrapper>
  );
};
