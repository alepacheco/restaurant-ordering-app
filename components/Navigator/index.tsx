import React from 'react';
import { createBottomTabNavigator } from 'react-navigation-tabs';
import { Settings } from '../../screens/Settings';
import { createStackNavigator } from 'react-navigation-stack';
import { RestaurantList } from '../../screens/RestaurantList';
import { Profile } from '../../screens/Profile';
import { Search } from '../../screens/Search';
import { RestaurantDetails } from '../../screens/RestaurantDetails';
import { LogIn } from '../../screens/Splash/LogIn';
import { SignUp } from '../../screens/Splash/SignUp';
import { Splash } from '../../screens/Splash';
import { Cart } from '../../screens/Cart';
import { Orders } from '../../screens/Orders';
import { ProductDetails } from '../../screens/ProductDetails';
import { TabBarIcon } from '../TabBar/TabBarIcon';
import { TabBar } from '../TabBar';
import { createAppContainer } from 'react-navigation';

const HomeTabIcon = (props: any) => (
  <TabBarIcon icon="home" focused={props.focused} />
);

const MapTabIcon = (props: any) => (
  <TabBarIcon icon="map" focused={props.focused} />
);

const ProfileTabIcon = (props: any) => (
  <TabBarIcon icon="profile" focused={props.focused} />
);

const BottomTabNavigator = createBottomTabNavigator(
  {
    Restaurants: {
      screen: RestaurantList,
      navigationOptions: {
        title: '',
        tabBarIcon: HomeTabIcon,
      },
    },
    Search: {
      screen: Search,
      navigationOptions: {
        title: '',
        tabBarIcon: MapTabIcon,
      },
    },
    Profile: {
      screen: Profile,
      navigationOptions: {
        title: '',
        tabBarIcon: ProfileTabIcon,
      },
    },
  },
  {
    tabBarComponent: TabBar,
  }
);

const SplashNavigator = createStackNavigator(
  {
    Splash: {
      screen: Splash,
      navigationOptions: {
        headerShown: false,
      },
    },
    LogIn: {
      screen: LogIn,
      navigationOptions: {
        headerShown: false,
      },
    },
    SignUp: {
      screen: SignUp,
      navigationOptions: {
        headerShown: false,
      },
    },
  },
  { initialRouteName: 'Splash' }
);

export const Navigator = createStackNavigator(
  {
    Home: {
      screen: BottomTabNavigator,
      navigationOptions: {
        headerShown: false,
      },
    },
    Settings,

    RestaurantDetails: {
      screen: RestaurantDetails,
      navigationOptions: {
        headerShown: false,
      },
    },
    ProductDetails: {
      screen: ProductDetails,
      navigationOptions: {
        headerShown: false,
      },
    },
    Cart: {
      screen: Cart,
      navigationOptions: {
        headerShown: false,
      },
    },
    Orders: {
      screen: Orders,
      navigationOptions: {
        headerShown: false,
      },
    },
    SplashNavigator: {
      screen: SplashNavigator,
      navigationOptions: {
        headerShown: false,
      },
    },
  },
  { initialRouteName: 'SplashNavigator' }
);

export const Router = createAppContainer(Navigator);
