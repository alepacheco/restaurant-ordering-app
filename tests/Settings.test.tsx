import React from 'react';
import renderer from 'react-test-renderer';
import { Theme } from '../components/Theme';
import { Settings } from '../screens/Settings';
jest.useFakeTimers();
describe('<Settings />', () => {
  it('renders correctly', async () => {
    let tree: null | renderer.ReactTestRenderer = null;

    await renderer.act(async () => {
      tree = renderer.create(
        <Theme>
          <Settings />
        </Theme>
      );
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
