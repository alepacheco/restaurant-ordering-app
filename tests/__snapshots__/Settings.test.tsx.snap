// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders correctly 1`] = `
<RNCAppearanceProvider
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "hsl(0, 0%, 90%)",
          "display": "flex",
        },
      ]
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 24,
              "fontWeight": "bold",
              "marginBottom": 12,
              "marginLeft": 12,
              "marginRight": 0,
              "marginTop": 18,
            },
          ]
        }
      >
        Need help?
      </Text>
    </RCTSafeAreaView>
  </View>
</RNCAppearanceProvider>
`;
